### ISSUE TITLE
Maximum Spanning Tree

### ISSUE THREAD

**Issue by chebee7i** [networkx/networkx#1563]
We have a minimum spanning tree, but it would be nice for users if we also had a maximum spanning tree. Yes, you can invert the weights, but that's a bit of a burden and wasteful when we can just change the sort order. Probably we can create a base function called optimum_spanning_edges() and then make the other two call it.

(Apologies for all the small issues...I can jump on them at some point soon, but I wanted to get them reported in case I forget)

**Comment by SanketDG** [comment]
What will the optimum_spanning_edges do? And will it be faster than reversing the sort order of Krusal's MST?

**Comment by chebee7i** [comment]
`optimum_spanning_edges` would do mostly what `minimum_spanning_edges` does now, except it would take a new keyword, opt (or something with a better name). If `opt` is `min`, then you do nothing different. If `opt` is `max`, then you pass `reverse=True` to the `sorted()` line.

This will be faster than reversing the weights, as that would require an additional pass through the graph. The outputs are not helpful at that point with respect to the original, unmodified graph. So the user would have to re-invert all the edges in the original graph. This requires two unnecessary extra passes through the edges of the graph.

**Comment by SanketDG** [comment]
Will `optimum_spanning_edges` be part of the package? Will it be imported into the namespace? I think not importing it should be okay. This is what I have in mind..
* Define `_optimum_spanning_edges`:
```python
def _optimum_spanning_edges(G, weight='weight', data=True, opt):
   # add if else checking for min or max value of opt
```python
* Then define `minimum_spanning_edges` and `maximum_spinning_edges` like this:
```python
def minimum_spanning_edges(G, weight='weight', data=True):
    _optimum_spanning_edges(G, weight='weight', data=True, opt='min')
```
```python
def maximum_spanning_edges(G, weight='weight', data=True):
    _optimum_spanning_edges(G, weight='weight', data=True, opt='max') 
```
Is this the right approach?

**Comment by jfinkels** [comment]
If there are only two options, the keyword argument should be a boolean, for example, `minimum=True`.

**Comment by SanketDG** [comment]
Yes, that makes sense.

**Comment by SanketDG** [comment]
This can be closed. Implemented in #1577.
